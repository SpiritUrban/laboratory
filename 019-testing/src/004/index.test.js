// npm install --save-dev jest-environment-jsdom

// module.exports = {
//   // ...остальные настройки
//   testEnvironment: "jsdom",
// };

// jest index -u  ---------- update snapshot!

const fs = require("fs");
const path = require("path");

test("simple-page snapshot", () => {
  // Чтение HTML-файла и его содержимого
  const htmlString = fs.readFileSync(
    path.resolve(__dirname, "./index.html"),
    "utf8"
  );

  // Устанавливаем HTML-строку в JSDOM
  document.documentElement.innerHTML = htmlString.toString();

  // Сохраняем снимок
  expect(document.documentElement).toMatchSnapshot();
});
