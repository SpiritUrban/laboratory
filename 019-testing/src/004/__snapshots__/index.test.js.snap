// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`simple-page snapshot 1`] = `
<html>
  <head>
    
  
    <title>
      Simple Page
    </title>
    
  
    <link
      href="style.css"
      rel="stylesheet"
    />
    

  </head>
  

  <body>
    
  
    <h1>
      Hello, World!
    </h1>
    
  
    <p>
      This is a simple page.
    </p>
    
  
    <button
      id="click-me"
    >
      Click Me!
    </button>
    



  </body>
</html>
`;
